### ISSUE TITLE
Cannot clear consumer configuration from plugins when using Cassandra

### ISSUE THREAD

**Issue by rainest** [Kong/kong#3308]
### Summary

Kong [allows PATCHes with parameters set to `null` to clear configuration](https://github.com/Kong/kong/commit/fc82dd73918bed332f46a66cdd8204d24e6e3fcb) where appropriate. Clearing consumers from a plugin in this fashion works when the database backend is Postgres, but not Cassandra.

### Steps To Reproduce

1. On a Cassandra-backed Kong, create a plugin associated with a consumer, e.g. `http -v post http://localhost:8001/plugins/ name="request-size-limiting" config.allowed_payload_size=128 consumer_id=<ID> api_id=<ID>`
2. Attempt to clear the consumer configuration, e.g. `http -v patch http://localhost:8001/plugins/<ID> consumer_id:=null`
3. Kong will respond a 500 and `"message": "An unexpected error occurred"`
4. Kong will log `2018/03/16 18:54:17 [error] 45#0: *2749 [lua] responses.lua:121: handler(): [Invalid] Invalid null value in condition for column id, client: 10.142.0.3, server: kong_admin, request: "PATCH /plugins/<ID> HTTP/1.1", host: "127.0.0.1:8001"`

### Additional Details & Logs

- Kong version: 0.12.3
- Kong debug logs: https://gist.github.com/rainest/5a40605678cb4f79429d7d19a956fa5a
- Operating System: standard Kong Docker image on GCE

Initial comment from @thibaultcha: Yep, there should be a check here to use the null uuid instead of NULL since in the C* table, `consumer_id` is part of the partition key iirc

**Comment by aledbf** [comment]
I get the same behavior with posgresql

**Comment by Tieske** [comment]
@aledbf with `0.13.0` I cannot reproduce it on postgres, only on cassandra

Using postgres:
```
vagrant@vagrant-ubuntu-trusty-64:/kong$ http POST :8001/plugins name=request-termination consumer_id=b19d7316-3fca-4507-9b3a-0e5bba59c9fd service_id=9152d097-ba50-4dcc-aa0e-fe6065491c57
HTTP/1.1 201 Created
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Wed, 28 Mar 2018 12:08:32 GMT
Server: kong/0.13.0
Transfer-Encoding: chunked

{
    "config": {
        "status_code": 503
    },
    "consumer_id": "b19d7316-3fca-4507-9b3a-0e5bba59c9fd",
    "created_at": 1522238913000,
    "enabled": true,
    "id": "0ca1aa0f-e537-4ce6-90c5-a72a66d663a6",
    "name": "request-termination",
    "service_id": "9152d097-ba50-4dcc-aa0e-fe6065491c57"
}

vagrant@vagrant-ubuntu-trusty-64:/kong$ http -v  name="request-size-limiting" config.allowed_payload_size=128 consumer_id=<ID> api_id=<ID>
-bash: syntax error near unexpected token `newline'
vagrant@vagrant-ubuntu-trusty-64:/kong$ http -v patch http://localhost:8001/plugins/0ca1aa0f-e537-4ce6-90c5-a72a66d663a6 consumer_id:=null
PATCH /plugins/0ca1aa0f-e537-4ce6-90c5-a72a66d663a6 HTTP/1.1
Accept: application/json
Accept-Encoding: gzip, deflate, compress
Content-Length: 21
Content-Type: application/json; charset=utf-8
Host: localhost:8001
User-Agent: HTTPie/0.8.0

{
    "consumer_id": null
}

HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Wed, 28 Mar 2018 12:20:45 GMT
Server: kong/0.13.0
Transfer-Encoding: chunked

{
    "config": {
        "status_code": 503
    },
    "created_at": 1522238913000,
    "enabled": true,
    "id": "0ca1aa0f-e537-4ce6-90c5-a72a66d663a6",
    "name": "request-termination",
    "service_id": "9152d097-ba50-4dcc-aa0e-fe6065491c57"
}

vagrant@vagrant-ubuntu-trusty-64:/kong$
```

Using cassandra:
```
vagrant@vagrant-ubuntu-trusty-64:/kong$ http POST :8001/plugins name=request-termination consumer_id=c5349cfb-1e94-4907-8b24-3232168e0413 service_id=289af09c-a075-40c5-b2a2-dfb15c143229
HTTP/1.1 201 Created
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Wed, 28 Mar 2018 12:28:48 GMT
Server: kong/0.13.0
Transfer-Encoding: chunked

{
    "config": {
        "status_code": 503
    },
    "consumer_id": "c5349cfb-1e94-4907-8b24-3232168e0413",
    "created_at": 1522240128604,
    "enabled": true,
    "id": "d565bff1-95ac-42bb-8242-a1ec401304f6",
    "name": "request-termination",
    "service_id": "289af09c-a075-40c5-b2a2-dfb15c143229"
}

vagrant@vagrant-ubuntu-trusty-64:/kong$ http -v PATCH http://localhost:8001/plugins/d565bff1-95ac-42bb-8242-a1ec401304f6 consumer_id:=null
PATCH /plugins/d565bff1-95ac-42bb-8242-a1ec401304f6 HTTP/1.1
Accept: application/json
Accept-Encoding: gzip, deflate, compress
Content-Length: 21
Content-Type: application/json; charset=utf-8
Host: localhost:8001
User-Agent: HTTPie/0.8.0

{
    "consumer_id": null
}

HTTP/1.1 500 Internal Server Error
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Date: Wed, 28 Mar 2018 12:29:45 GMT
Server: kong/0.13.0
Transfer-Encoding: chunked

{
    "message": "An unexpected error occurred"
}
```

and the error is:

> 2018/03/28 12:29:45 [error] 30407#0: *365 [lua] responses.lua:121: handler(): [Invalid] Invalid null value in condition for column id, client: 127.0.0.1, server: kong_admin, request: "PATCH /plugins/d565bff1-95ac-42bb-8242-a1ec401304f6 HTTP/1.1", host: "localhost:8001"

**Comment by thibaultcha** [comment]
On Postgres the HTTP 500 error is triggered by a different code path that with Cassandra. This is the log you'd get when trying to unset `service_id` (`route_id` would be similar):
```
2018/04/03 19:34:49 [error] 11962#0: *4 lua coroutine: runtime error: ./kong/dao/schemas/plugins.lua:101: attempt to concatenate field 'service_id' (a userdata value)
stack traceback:
coroutine 0:
        ./kong/dao/schemas/plugins.lua: in function 'self_check'
        ./kong/dao/schemas_validation.lua:289: in function 'validate'
        ./kong/dao/model_factory.lua:27: in function 'validate'
        ./kong/dao/dao.lua:301: in function 'update'
        ./kong/api/crud_helpers.lua:213: in function 'patch'
        ./kong/api/routes/plugins.lua:70: in function <./kong/api/routes/plugins.lua:69>
```

@bungle came up with an appropriate fix I am merging as of right now in #3355
